app/data_model/answer_store.py

```diff
@@ -206,16 +206,32 @@ def map(self, group_id=None, block_id=None, answer_id=None, group_instance=None,
 
         return OrderedDict(sorted(result.items(), key=lambda t: natural_order(t[0])))
 
-    def remove(self, answer):
+    def remove_answer(self, answer):
         """
         Removes an answer from the answer store.
 
         :param answer: A dict of flattened answer details.
         """
         index = self.find(answer)
-        if index:
+        if index is not None:
             del self.answers[index]
 
+    def remove(self, group_id=None, block_id=None, answer_id=None, group_instance=None, answer_instance=None):
+        """
+        Removes answer(s) from the answer store.
+
+        :param answer_id:
+        :param block_id:
+        :param group_id:
+        :param answer_instance:
+        :param group_instance:
+        """
+        use_filter = (group_id or block_id or answer_id or group_instance or answer_instance) is not None
```

Comment:
Good point. I've moved the `use_filter` logic out of the `remove` and `map` methods. I think we should consider making the `filter` method something we can chain in the future, so we could have something like `answer_store.filter(group_id='group1').remove()`